libs/components/indicators/src/lib/modules/label/label.component.ts

```diff
@@ -53,4 +97,43 @@ export class SkyLabelComponent {
     this.baseIcon = indicatorIcon.modernThemeBaseIcon;
     this.topIcon = indicatorIcon.modernThemeTopIcon;
   }
+
+  #updateDescriptionComputed(): void {
+    if (this.descriptionType) {
+      switch (this.descriptionType) {
+        case 'none':
+          this.descriptionComputed = undefined;
+          break;
+        case 'custom':
+          this.descriptionComputed = this.customDescription;
+          break;
+        default:
+          this.#unsubscribe();
+
+          this.#currentSub = this.resources
+            .getString(
+              'skyux_label_sr_' + this.descriptionType.replace(/-/g, '_')
+            )
+            .subscribe((value) => {
+              this.descriptionComputed = value;
+              this.changeDetector.markForCheck();
+            });
+
+          break;
+      }
+    } else {
+      this.descriptionComputed = undefined;
+    }
+  }
+
+  #unsubscribe(): void {
+    if (this.#currentSub) {
+      this.#currentSub.unsubscribe();
+      this.#currentSub = undefined;
+    }
+  }
+
+  public ngOnDestroy(): void {
```

Comment:
Public methods should be above private methods. We generally put Angular lifecycle methods directly after the `constructor` so they're easy to find.